user-interface/src/main/java/life/qbic/datamanager/views/projects/create/ProjectDesignLayout.java

```diff
@@ -97,16 +128,32 @@ private void initCodeGenerationButton() {
     generateCodeButton.setTooltipText("Generate Project ID");
     generateCodeButton.addThemeVariants(ButtonVariant.LUMO_ICON);
     generateCodeButton.addClickListener(
-        buttonClickEvent -> codeField.setValue(ProjectCode.random().value()));
+        //Only non-used project code should be provided
+        buttonClickEvent -> {
+          String newProjectCode = ProjectCode.random().value();
+          boolean isProjectCodeDuplicated = !isProjectCodeUnique(newProjectCode);
+          while (isProjectCodeDuplicated) {
+            newProjectCode = ProjectCode.random().value();
+            isProjectCodeDuplicated = isProjectCodeUnique(newProjectCode);
+          }
```

Comment:
Very NIT: maybe consider introducing a control on how often this can run. For example try to find it 10 times and if nothing was found log an error and ask the user to click the button again?
In theory this only becomes while(true) if there are no project codes left but I think depending on how many project codes there are it might be good to have a fail-safe. (This way the users will probably let us know when project codes seem to run out. Which will not be the case for quite a while thus very NIT)

Reply:
another idea would be to simply generate a random code and display if it is already taken. so the user can click the button again :shrug:

Reply:
Good idea 👍